public/js/analytics.js

```diff
@@ -0,0 +1,82 @@
+const getAnalytics = async (county) => {
+  const pm1Elem = document.querySelector('.pm1-range .title');
+  const pm2Elem = document.querySelector('.pm2-5-range .title');
+  const pm0Elem = document.querySelector('.pm10-range .title');
+  const noiseElem = document.querySelector('.noise-range .title');
+
+  pm1Elem.innerHTML = 0;
+  pm2Elem.innerHTML = '0';
+  pm0Elem.innerHTML = '0';
+  noiseElem.innerHTML = 0;
+
+  const rangeMap = {
+    pm1: { high: 0, low: 0 },
+    pm2: { high: 0, low: 0 },
+    pm0: { high: 0, low: 0 },
+    noise: { high: 0, low: 0 },
+  };
+
+  const getValue = (sensorReading) => {
+    const updateMap = (type, data) => {
+      rangeMap[type].high =
+        Number(data.value) > rangeMap[type].high
+          ? Number(data.value)
+          : rangeMap[type].high;
+      if (rangeMap[type].low === 0) {
+        rangeMap[type].low = Number(data.value);
+      } else {
+        rangeMap[type].low =
+          Number(data.value) < rangeMap[type].low
+            ? Number(data.value)
+            : rangeMap[type].low;
+      }
+    };
+    sensorReading.forEach((reading) => {
+      reading.sensordatavalues.forEach((data) => {
+        switch (data.value_type) {
+          case 'P1':
+            updateMap('pm1', data);
+            break;
+          case 'P2':
+            updateMap('pm2', data);
+            break;
+          case 'P0':
+            updateMap('pm0', data);
+            break;
+          case 'noise_Leq':
+            updateMap('noise', data);
+            break;
+          default:
+            break;
+        }
+      });
+    });
+  };
+  const nodes = await window.aq.getNodes();
+  const counties = await window.sheets.getCityCountyMap();
+
+  const selectedCounty = nodes.filter((node) => {
+    return counties[node.location.city] === county;
+  });
+
+  const sensorData = selectedCounty
+    ?.map((node) => node.sensors)
+    ?.map((data) => data.map((result) => result.sensordatas));
```

Comment:
```suggestion
  const sensorDatas = selectedCounty
    ?.flatMap((node) => node.sensors)
    ?.flatMap((data) => data.map((result) => result.sensordatas));
```

[flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)

Reply:
Done